src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,97 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+import { validationPattern } from '../../utils/urlValidationPattern';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+const initialMovie = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [movie, setMovie] = useState<Movie>(initialMovie);
+  const {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  } = movie;
+  const isEmpty = !title.trim()
+  || !imdbId.trim()
+  || !imdbUrl.trim()
+  || !imgUrl.trim();
+  const isPassingValidation = validationPattern.test(imdbUrl)
+    && validationPattern.test(imgUrl);
+
+  const handleChangeFieldValue = (
+    fieldName: keyof Movie,
+    fieldValue: string,
+  ) => {
+    setMovie((prev) => ({ ...prev, [fieldName]: fieldValue }));
+  };
```

Comment:
👍 But we can do it even simplier

```suggestion
  const handleChangeFieldValue = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = event.target;
    
    setMovie((prev) => ({
      ...prev,
      [name]: value
    }));
  };
```